### ISSUE TITLE
Task fails when VS2017 RC is installed

### ISSUE THREAD

**Issue by skolima** [stevewillcock/grunt-msbuild#38]
When you install the current RC for VS2017, grunt-msbuild fails to find the msbuild executable with message:

`Fatal error: Unrecognised MSBuild version "15"`

This does not depend on any changes to the project files, just on the new VS being installed on the machine.

The actual path depends on the version of VS2017 installed:
`C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin`

**Comment by leopignataro** [comment]
Same problem here. This thread seems to be related:

http://stackoverflow.com/questions/40694598/how-do-i-call-visual-studio-2017-rcs-version-of-msbuild-from-a-bat-file

BTW, VS 2017 RTM is less than 2 weeks away now (launch is on March 7th)

**Comment by stevewillcock** [comment]
Thanks for the reports - I'll try to take a look at this on the weekend.

**Comment by sherland** [comment]
Error with release-version.

The problem occurs when the version is not specified. Then msbuild.js createCommand will try to get newest version in {ProgramFiles(x86)}\MSBuild based on file-name. On my PC this will be 15.0 - which is not  defined in the versions array.

Since locating msbuild.exe is more complex now, a temporary workaround could be to check if the "resolved" version is 15.0, and change it to 14.0 if it is? Then existing projects would at least build without change after installing VS2017.

BTW: My fairly big project was building successfully when I added version: 14.0 to the options in the gruntfile

**Comment by sherland** [comment]
About locating msbuild.exe:
https://github.com/Microsoft/vswhere can perhaps be included? They have a usage example:
```
@echo off

for /f "usebackq tokens=1* delims=: " %%i in (`vswhere -latest -requires Microsoft.Component.MSBuild`) do (
  if /i "%%i"=="installationPath" set InstallDir=%%j
)

if exist "%InstallDir%\MSBuild\15.0\Bin\MSBuild.exe" (
  "%InstallDir%\MSBuild\15.0\Bin\MSBuild.exe" %*
)
```

**Comment by stevewillcock** [comment]
I've pushed a fix for this as version 0.4.0 - can you let me know if you are still having problems after using this version? Thanks!

**Comment by sherland** [comment]
Works for me! (Sorry for not responding quicker.)

BTW: I have issues with building projects that contain "Dependency Validation" projects, but this is not related to grunt-msbuild since the issue is the same when I use msbuild.exe directly.

**Comment by stevewillcock** [comment]
Great, thanks for the feedback :)

**Comment by recaro** [comment]
trying out the 0.4.0 version failed for me with the following:
`00:16:30.966 'C:\Program' is not recognized as an internal or external command,`
`00:16:30.966 operable program or batch file.`
`00:16:30.966 [33mWarning: Command failed: C:\Program Files (x86)\Jenkins\workspace\Build - Release\BE\node_modules\grunt-msbuild\bin\vswhere.exe -latest -requires Microsoft.Component.MSBuild`
`00:16:30.966 'C:\Program' is not recognized as an internal or external command,`
`00:16:30.966 operable program or batch file.`

It seems the `vswhere.exe` command is not properly quoted for spaces.

**Comment by stevewillcock** [comment]
I've published a fix for the space-in-pathnames issue as 0.4.1

**Comment by stevewillcock** [comment]
Closing this as I think it's now fixed. If you are still having issues after installing 0.4.2 then please let me know.
